### PR TITLE
Introduce Compass split proposal

### PR DESCRIPTION
**Pull request by pkosiec**
**Description**

Changes proposed in this pull request:

- Introduce Compass split proposal

### CODE REVIEW

File: docs/internal/proposals/compass-split/compass-split.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+## Reasons
+
+There are two main reasons for the Compass project split:
+
+- Separation of concerns: Application Connectivity components are separated from Runtime-related components.
+- Change of the Application Connectivity components ownership.
+
+## Implementation
+
+The Compass project is split into two repositories:
```
Comment (crabtree) [inline docs/internal/proposals/compass-split/compass-split.md:20]: I think we should also mention here the split of the actual schema migrator component. Now it holds the migrations for the Director, Provisioner, and KEB.

File: docs/internal/proposals/compass-split/compass-split.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  - Watch KCP repo for changes
+  - Install Compass from fixed `COMPASS_VERSION` from KCP repo
+  - Modify from where we get Kyma version to install: Install Kyma from fixed `KYMA_VERSION` from KCP repo
+  - Install KCP installer as a new step
+
+### Migration of existing Compass installations
+
+Migration of the existing Compass installation doesn't require any custom operations.
+
+Once a pipeline is modified to install Kyma Control Plane components from `control-plane` repository on top of existing Compass installation, the transition period begins. In the transition period, there will be duplicated KCP components on the cluster. They will be deployed in two different namespaces: `compass-system` and `kcp-system`. The transition period ends once the Compass chart is modified and all KCP components are deleted from the Compass chart.
```
Comment (klaudiagrz) [inline docs/internal/proposals/compass-split/compass-split.md:104]:
```suggestion
Once a pipeline is modified to install Kyma Control Plane components from the `control-plane` repository on top of the existing Compass installation, the transition period begins. In the transition period, there will be duplicated KCP components on the cluster. They will be deployed in two different Namespaces: `compass-system` and `kcp-system`. The transition period ends when the Compass chart is modified and all KCP components are deleted from the Compass chart.
```

File: docs/internal/proposals/compass-split/compass-split.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+### Cleanup
+
+Once Compass components are migrated to a separate internal environment:
```
Comment (klaudiagrz) [inline docs/internal/proposals/compass-split/compass-split.md:108]:
```suggestion
Once the Compass components are migrated to a separate internal environment:
```

File: docs/internal/proposals/compass-split/compass-split.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+- Remove Compass installation step from internal environment pipelines
```
Comment (klaudiagrz) [inline docs/internal/proposals/compass-split/compass-split.md:110]:
```suggestion
- Remove the Compass installation step from internal environment pipelines.
```

File: docs/internal/proposals/compass-split/compass-split.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+- Remove Registration Job in Kyma Control Plane
```
Comment (klaudiagrz) [inline docs/internal/proposals/compass-split/compass-split.md:111]:
```suggestion
- Remove the registration job from Kyma Control Plane.
```

**Pull request merged**